Mathlib/Dynamics/PeriodicPts/Lemmas.lean

```diff
@@ -63,6 +65,54 @@ theorem Commute.minimalPeriod_of_comp_eq_mul_of_coprime {g : α → α} (h : Com
   · exact (isPeriodicPt_minimalPeriod _ _).const_mul _
   · exact (isPeriodicPt_minimalPeriod _ _).mul_const _
 
+section Fintype
+
+variable [Fintype α]
+
+open Fintype
+
+theorem minimalPeriod_le_card : minimalPeriod f x ≤ card α := by
+  rw [← periodicOrbit_length]
+  exact List.Nodup.length_le_card nodup_periodicOrbit
+
+theorem isPeriodicPt_factorial_card_of_mem_periodicPts (h : x ∈ periodicPts f) :
+    IsPeriodicPt f (card α)! x :=
+  isPeriodicPt_iff_minimalPeriod_dvd.mpr
+    (Nat.dvd_factorial (minimalPeriod_pos_of_mem_periodicPts h) minimalPeriod_le_card)
+
+theorem mem_periodicPts_iff_isPeriodicPt_factorial_card :
+    x ∈ periodicPts f ↔ IsPeriodicPt f (card α)! x :=
+  ⟨isPeriodicPt_factorial_card_of_mem_periodicPts, fun h ↦
+   minimalPeriod_pos_iff_mem_periodicPts.mp
+    (IsPeriodicPt.minimalPeriod_pos (Nat.factorial_pos _) h)⟩
```

Comment:
I think this would be more readable as
```suggestion
    x ∈ periodicPts f ↔ IsPeriodicPt f (card α)! x where
  mp :=isPeriodicPt_factorial_card_of_mem_periodicPts
  mpr h := minimalPeriod_pos_iff_mem_periodicPts.mp
    (IsPeriodicPt.minimalPeriod_pos (Nat.factorial_pos _) h)
```